### PR TITLE
udp receiver for color

### PR DESCRIPTION
**Pull request by Rylan-Meilutis**
Receiver for the color data being sent by the roborio

### CODE REVIEW

File: Vision2022/main.py [hunk 1]
```diff
@@ -170,23 +175,31 @@ def send(data):
...
-    wait_for_other_robot)
+        hoop_x, hoop_y, hoop_angle, ball_x, ball_y, ball_angle, int(ball_found), int(ball_found),
+        wait_for_other_robot)
     send(data)
 
 
+def receive():
+    global color
+    global sock
+    data, addr = sock.recvfrom(1024)  # buffer size is 1024 bytes
```
Comment (mmarchetti) [inline Vision2022/main.py]: Need to handle the case where no data is returned (I think it returns None; a test with no sender will demonstrate). Also this returns bytes and not a string, so have to call decode() on it.

Comment (Rylan-Meilutis) [reply Vision2022/main.py]: Should be fixed

File: Vision2022/main.py [hunk 2]
```diff
@@ -170,23 +175,32 @@ def send(data):
     s.sendto(data.encode('utf-8'), address)
 
 
-def send_data(hoop_found, hoop_x, hoop_y, hoop_angle, ball_found, ball_x, ball_y, ball_angle, wait_for_other_robot):
+def send_data(hoop_x, hoop_y, hoop_angle, ball_found, ball_x, ball_y, ball_angle, wait_for_other_robot):
+
     data = '%3.1f %3.1f %3.1f %3.1f %3.1f %3.1f %d %d %d \n' % (
-    hoop_x, hoop_y, hoop_angle, ball_x, ball_y, ball_angle, int(ball_found), int(ball_found),
-    wait_for_other_robot)
+        hoop_x, hoop_y, hoop_angle, ball_x, ball_y, ball_angle, int(ball_found), int(ball_found),
```
Comment (mmarchetti) [inline Vision2022/main.py]: This doesn't seem right - we're returning `ball_found` twice but not `hoop_found`.

Comment (Rylan-Meilutis) [reply Vision2022/main.py]: fixed

**Review by mmarchetti — DISMISSED** [review]

**Comment by mmarchetti** [comment]
We were relying on blocking to match the viewer loop to the sender's frame rate. I think that the way this is currently, we will flash to the blank placeholder frames between normal frames. Instead of setting the socket to be nonblocking, consider setting a socket timeout of 300ms or so. Then it will try for a bit before giving up and falling back to the placeholder (the receive call will raise a timeout error, which your code is already set up to catch and set data=None).

**Review by mmarchetti — APPROVED** [review]

**Pull request merged**